### PR TITLE
[Feature] add log collector

### PR DESCRIPTION
**Pull request by RockeyCoss**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add a develop tool: Log Collector

## Modification

Add a develop tool: Log Collector

### CODE REVIEW

File: .dev/log_collector/example_config.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+work_dir = '../mmsegmentation-master/work_dirs'
+metric = 'mIoU'
+# Don't specify the log_items and ignore_keywords at the same time
+log_items = [
+    'segformer_mit-b5_512x512_160k_ade20k_cnn_lr_with_warmup',
+    'segformer_mit-b5_512x512_160k_ade20k_cnn_no_wramup_lr',
```
Comment (MeowZheng) [inline .dev/log_collector/example_config.py:6]: typo: wramup -> warmup

File: .dev/log_collector/readme.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+```python
+# The work directory that contains folders that contains .log.json files
+work_dir = '../mmsegmentation-master/work_dirs'
+# The metric used to find the best evaluation
+metric = 'mIoU'
+# Don't specify the log_items and ignore_keywords at the same time
+# The folders that contain the logs you want to collect
+# The folders should be the subdirectories of work_dir
+log_items = ['segformer_mit-b5_512x512_160k_ade20k_cnn_lr_with_warmup',
+                 'segformer_mit-b5_512x512_160k_ade20k_cnn_no_wramup_lr',
```
Comment (MeowZheng) [inline .dev/log_collector/readme.md:26]: typo: wramup -> warmup

File: .dev/log_collector/readme.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+other_info_keys = ['other_key']
+# The output markdown file's name
+markdown_file ='markdowns/lr_in_trans.json.md'
+# The output json file's name（optional）
+json_file = 'trans_in_cnn.json'
+```
+
+ The structure of the work-dir directory should be like：
+
+```
```
Comment (MeowZheng) [inline .dev/log_collector/readme.md:45]: Fenced code blocks should have a language specified
```suggestion
```text
```

File: .dev/log_collector/readme.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        │   ├── folder1
+        │   │   ├── time1.log.json
+        │   │   ├── time2.log.json
+        │   │   ├── time3.log.json
+        │   │   ├── time4.log.json
+        │   ├── folder2
+        │   │   ├── time5.log.json
+        │   │   ├── time6.log.json
+        │   │   ├── time7.log.json
+        │   │   ├── time8.log.json
```
Comment (MeowZheng) [inline .dev/log_collector/readme.md:56]: why is there indentation?

File: .dev/log_collector/readme.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+```
+
+Then , you can run log_collector.py by using command:
+
+```bash
+python log_collector.py ./example_config.py
+```
+
+
+
```
Comment (MeowZheng) [inline .dev/log_collector/readme.md:66]: remove blank lines, only one blank line is enough

File: .dev/log_collector/example_config.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    'segformer_mit-b5_512x512_160k_ade20k_mit_trans_lr',
+    'segformer_mit-b5_512x512_160k_ade20k_swin_trans_lr'
+]
+# or
+ignore_keywords = ['segformer']
```
Comment (MeowZheng) [inline .dev/log_collector/example_config.py:11]: if `log_items` and `ignore_keywords` cannot be specified at the same time, I suggest:
```suggestion
# or only set ignore_keywords, here is an example
# the folder name containing `'segformer'` won't be collected
# ignore_keywords = ['segformer']
```

**Review by MeowZheng — COMMENTED** [review]
add copyright

File: .dev/log_collector/example_config.py (hunk 6 above)
Comment (MeowZheng) [inline .dev/log_collector/example_config.py:11]: if `log_items` and `ignore_keywords` cannot be specified at the same time, why give values to `log_items` and `ignore_keywords` in `example_config.py`. I suggest:
```suggestion
# the folder containing `'segformer'` won't be collected
ignore_keywords = ['segformer']

# or specify the log files we would like to collect in `log_items`
# log_items = [
#     'segformer_mit-b5_512x512_160k_ade20k_cnn_lr_with_warmup',
#     'segformer_mit-b5_512x512_160k_ade20k_cnn_no_wramup_lr',
#     'segformer_mit-b5_512x512_160k_ade20k_mit_trans_lr',
#     'segformer_mit-b5_512x512_160k_ade20k_swin_trans_lr'
# ]
```

**Review by MeowZheng — APPROVED** [review]
LGTM

File: .dev/log_collector/readme.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
+# Log Collector
+
+## Function
+
+Automatically collect logs and write the result in a json file or markdown file.
+
+If there are several .log.json files in one folder, Log Collector assumes that the .log.json files other than the first one is resume from the preceding .log.json file. Log Collector returns the result considering all .log.json files.
```
Comment (Junjun2016) [inline .dev/log_collector/readme.md:7]:
```suggestion
If there are several `.log.json` files in one folder, Log Collector assumes that the `.log.json` files other than the first one are resume from the preceding `.log.json` file. Log Collector returns the result considering all `.log.json` files.
```

**Comment by Junjun2016** [comment]
Hi @MengzhangLI 
Please review this PR and give some comments.

**Comment by MengzhangLI** [comment]
This enhancement could be ran correctly.

But the metric of all logs should keep the same, otherwise, the error may happens.

For example, if metric of `log.json` files have `mDice` and `mIoU` at the same time, but the input is `mDice`, then error may be raised during handling certain `log.json` file whose metric is `mIoU`.

image


However, if all `.log.json` files have same metric, it could run correctly.

image


## lr_in_trans.json.md

|exp_num|method|mDice best|best index|mDice last|last index|last iter num|
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
|1|deeplabv3_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_chase-db1|0.8949|10|0.8949|10|40000|
|2|deeplabv3_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_stare|0.9065|10|0.9065|10|40000|
|3|deeplabv3_unet_s5-d16_ce-1.0-dice-3.0_256x256_40k_hrf|0.8961|8|0.8956|10|40000|
|4|deeplabv3_unet_s5-d16_ce-1.0-dice-3.0_64x64_40k_drive|0.8892|7|0.8884|10|40000|
|5|fcn_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_chase-db1|0.8952|10|0.8952|10|40000|
|6|fcn_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_stare|0.9065|10|0.9065|10|40000|
|7|fcn_unet_s5-d16_ce-1.0-dice-3.0_256x256_40k_hrf|0.8964|10|0.8964|10|40000|
|8|fcn_unet_s5-d16_ce-1.0-dice-3.0_64x64_40k_drive|0.8897|4|0.8871|10|40000|
|9|pspnet_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_chase-db1|0.8945|10|0.8945|10|40000|
|10|pspnet_unet_s5-d16_ce-1.0-dice-3.0_128x128_40k_stare|0.9072|10|0.9072|10|40000|
|11|pspnet_unet_s5-d16_ce-1.0-dice-3.0_256x256_40k_hrf|0.8969|10|0.8969|10|40000|
|12|pspnet_unet_s5-d16_ce-1.0-dice-3.0_64x64_40k_drive|0.889|4|0.8876|10|40000|

**Review by MengzhangLI — APPROVED** [review]

**Review by Junjun2016 — APPROVED** [review]

**Pull request merged**